services/web/server/src/simcore_service_webserver/garbage_collector/_core_disconnected.py

```diff
@@ -0,0 +1,167 @@
+import logging
+
+from aiohttp import web
+from redis.asyncio import Redis
+from servicelib.common_headers import UNDEFINED_DEFAULT_SIMCORE_USER_AGENT_VALUE
+from servicelib.utils import logged_gather
+
+from ..projects.exceptions import ProjectLockError, ProjectNotFoundError
+from ..projects.projects_api import remove_project_dynamic_services
+from ..redis import get_redis_lock_manager_client
+from ..resource_manager.registry import RedisResourceRegistry, ResourcesDict
+from ._core_guests import remove_guest_user_with_all_its_resources
+from .settings import GUEST_USER_RC_LOCK_FORMAT
+
+_logger = logging.getLogger(__name__)
+
+
+async def remove_disconnected_user_resources(
+    registry: RedisResourceRegistry, app: web.Application
+) -> None:
+    lock_manager: Redis = get_redis_lock_manager_client(app)
+
+    #
+    # In redis jargon, every entry is denoted as "key"
+    #   - A key can contain one or more fields: name-value pairs
+    #   - A key can have a limited livespan by setting the Time-to-live (TTL) which
+    #       is automatically decreasing
+    #
+    # - Every user can open multiple sessions (e.g. in different tabs and/or browser) and
+    #   each session is hierarchically represented in the redis registry with two keys:
+    #     - "alive" that keeps a TLL
+    #     - "resources" to keep a list of resources
+    # - A resource is defined as something that can be acquire/released and in some times
+    #   also shared. For instance, websocket_id, project_id are resource ids. The first is established
+    #   between the web-client and the backend.
+    #
```

Comment:
yes you know about it :-)  I guess the explanation might be misleading. This refers to the redis `resources` db. When a user session (i.e. user_id+client_session_id) acquires a resource, it gets noted there. For the moment a resource in this context is defined as a socket-id and a project-id. Nothing else.

Reply:
Yes. At the time it seems we envision the possibility of sharing project_id (which is something that will definitively happen) . 
Actually if you look at the code below `other_keys_with_this_resource` collects other user_sessions that also might have acquired the resource.

One of the big design problems of this piece is that it uses a low level access to redis (i.e. `resource_manager.registry`) instead of `resource_manager.user_sessions`. That is why at the logic is so convoluted and after all this time it gets even more difficult to understand.